src/stransport_stream.c

```diff
@@ -230,7 +230,8 @@ static void stransport_free(git_stream *stream)
 	stransport_stream *st = (stransport_stream *) stream;
 
 	git_stream_free(st->io);
-	CFRelease(st->ctx);
+	if (st->ctx)
+		CFRelease(st->ctx);
```

Comment:
It's either this or the other block. With this patch we avoid calling the free function when it's `NULL` so it's either one or the other.